File: Java/Sorting/bubble_sort.java
import java.util.Scanner;

class bubble_sort {
    public static void main(String args[]) {
        int num;
        int temp = 0;
        try (Scanner s = new Scanner(System.in)) {
            System.out.print("ENTER THE NUMBER OF ELEMENTS: ");

            num = s.nextInt();

            int[] array = new int[num];
            System.out.println("ENTER THE ELEMENTS OF ARRAY: ");
            for (int i = 0; i < num; i++) {
                array[i] = s.nextInt();
            }
            System.out.println("UNSORTED ARRAY: ");

            for (int i = 0; i < num; i++) {
                System.out.println(array[i]);
            }

            for (int i = 0; i < num; i++) {
                for (int j = 1; j < (num - i); j++) {
                    if (array[j - 1] > array[j]) {

                        temp = array[j - 1];
                        array[j - 1] = array[j];
                        array[j] = temp;
                    }

                }
            }
            System.out.println("SORTED ARRAY: ");
            for (int i = 0; i < num; i++) {
                System.out.println(array[i]);
            }
        }

    }

}

File: C/Sorting/bubble_sort.c
#include <stdio.h>

int main()
{
    int a[20], n, i, j, temp, k, s;

    
    printf("ENTER THE SIZE OF ARRAY: ");
    scanf("%d", &n);

    for (i = 0; i < n; i++)
    {
        printf("ENTER THE ELEMENTS: ");
        scanf("%d", &a[i]);
    }

    for (s = 0; s < n - 1; s++)
    {

        for (k = 0; k < n - 1; k++)
        {
            if (a[k] > a[k + 1])
            {
                temp = a[k];
                a[k] = a[k + 1];
                a[k + 1] = temp;
            }
        }
    }

    for (j = 0; j < n; j++)
    {
        printf("%d\t", a[j]);
    }
}